### PR TITLE
Atempt to synchronize with main branch and add my code

### PR DESCRIPTION
**Pull request by dronecz**
Hi, 

I rewrote the WiFi setup code and manually synced my code with the latest code in the repository. 
This was tested on ES3ink board with SHT40 and battery connected. 

image

### CODE REVIEW

**Comment by andy9a9** [comment]
Uff, 29 commits, please try to squash some commits....

File: src/fonts/OpenSansSB_24px.h [hunk 1]
```diff
@@ -0,0 +1,199 @@
+// Created by http://oleddisplay.squix.ch/ Consider a donation
```
Comment (andy9a9) [inline src/fonts/OpenSansSB_24px.h:1]: Isn't possible to re-use some parts from previous fonts and make a common out of it?

File: src/main.cpp [hunk 2]
```diff
@@ -200,51 +374,253 @@ uint64_t deepSleepTime = defaultDeepSleepTime; // actual sleep time in minutes,
...
+  // Set network name to wi-fi mac address
+  String hostname = "INK_";
+  hostname += WiFi.macAddress();
+  // Replace colon with nothing
+  hostname.replace(":", "");
+
+  displayInit();
+  delay(500);
+
+  if (DISPLAY_RESOLUTION_X >= 800) {
```
Comment (andy9a9) [inline src/main.cpp:539]: I guess a separate function will be better

File: src/main.cpp [hunk 3]
```diff
@@ -200,51 +374,253 @@ uint64_t deepSleepTime = defaultDeepSleepTime; // actual sleep time in minutes,
...
+
+  */
+
+  String qrString = "WIFI:S:";
+  qrString += hostname;
+  qrString += ";T:WPA;P:zivyobraz;;";
+  //Serial.println(qrString);
+
+  if (DISPLAY_RESOLUTION_X >= 800) {
+    drawQrCode(qrString.c_str(), 4, (DISPLAY_RESOLUTION_Y / 2) + 50, DISPLAY_RESOLUTION_X / 4, 4);
```
Comment (andy9a9) [inline src/main.cpp:600]: Same as above

**Comment by MultiTricker** [comment]
Don’t do anything, I’m like 80% done 😅😂 Hopefully finnish tomorrow evening.

File: src/main.cpp (hunk 3 above)
Comment (MultiTricker) [reply src/main.cpp:600]: I already found some things you are pointing out. This is chaos for me - I need to deal with conflicts first, merge this PR and then maybe improve code. Right now, I’m dealing with too many things at once.

**Comment by MultiTricker** [comment]
I did my best to manually check both code bases and import changes from @dronecz development. There might be some issues - I need to try compile the code and test it on both boards and some displays, add another Wi-Fi configuration screen layouts etc.

When I have working code and will be able to release 2.1 firmware which also solves problem with buffer reading on slow connections (and I really wanted to release this like week ago!), then release will be issued.

After that, any code improvements are welcome, but (as you already know ;)) I have very little time for ZivyObraz and need to put my work elsewhere.

**Pull request merged**